river/tree/_attribute_observer/numeric_attribute_class_observer_gaussian.py

```diff
@@ -11,14 +11,20 @@
 class NumericAttributeClassObserverGaussian(AttributeObserver):
     """Numeric attribute observer for classification tasks that is based on
     Gaussian estimators.
+
+
+    Parameters
+    ----------
+    n_splits
+        The number of partitions to consider when querying for split candidates.
     """
 
-    def __init__(self, n_bins: int = 10):
+    def __init__(self, n_splits: int = 10):
```

Comment:
`n_bins` is the standard way to refer to this parameter as it makes reference to "bins" where observations will be allocated. This is the way I have seen it named in multiple ml packages. However, I am open to discussing this.

Reply:
I am not against using bins for the name. It makes sense to keep the current standards adopted by the research community.

I changed its name because it might confuse the user when comparing this AO with the Histogram one. The newly added Histogram AO has a parameter `n_bins` with a different purpose.

Reply:
To some extent, the meaning of "bins" is similar in both AO. I don't think this would be an issue.